### ISSUE TITLE
Optimize multiple models at once

### ISSUE THREAD

**Issue by noahtren** [google-research/flax#28]
Curious how I can use Flax to optimize two models with the same loss function.

I have an encoder-decoder setup where I'm using [teacher forcing](https://machinelearningmastery.com/teacher-forcing-for-recurrent-neural-networks/) to train a decoder LSTM. This requires doing one encoding step of an input, then a lot of incremental decoding steps where loss is added after each one. I plan on doing this by having a distinct encoder and decoder network. Any quick intuitions for how I can get Flax to optimize both simultaneously? It looks like optimizers only take in a single object to optimize.

Thanks!

**Comment by noahtren** [comment]
One idea is to define a single loss function that takes both models as an input, and then use two optimizers to optimize either model while holding the other constant, like so:
```python
e_optim, e_loss, e_aux = e_optim.optimize(lambda e: loss_fn(e, decoder))
d_optim, d_loss, d_aux = d_optim.optimize(lambda d: loss_fn(encoder, d))
```
One problem is that this would essentially double training time.

Just thinking out loud and hoping to see if there's a better way to do this with Flax.

**Comment by noahtren** [comment]
^^ that didn't work. Realized I can just output the loss directly from my model, so I did that instead.
